src/renderer/store/actions.js

```diff
@@ -476,13 +476,29 @@ async function fetchTradeHistory({ state, commit }, { marketName, isRequestSilen
 }
 
 async function fetchOrderHistory({ commit }, { isRequestSilent }) {
-  let orders;
+  let orderHistory;
   commit(isRequestSilent ? 'startSilentRequest' : 'startRequest',
     { identifier: 'fetchOrderHistory' });
 
+  const currentNetwork = network.getSelectedNetwork();
+  const currentWallet = wallets.getCurrentWallet();
+  const orderHistoryStorageKey = `orderhistory.${currentWallet.address}.${currentNetwork.net}`;
   try {
-    orders = await dex.fetchOrderHistory();
-    commit('setOrderHistory', orders);
+    orderHistory = await fetchCachedData(orderHistoryStorageKey);
+    commit('setOrderHistory', orderHistory);
+  } catch (holdings) {
+    commit('setOrderHistory', orderHistory);
+  }
+
+  try {
+    if (orderHistory && orderHistory.length > 0) {
+      const newOrders = await dex.fetchOrderHistory(0, orderHistory[0].created);
```

Comment:
For this to work it would need to use lastUpdated instead of created or it won’t get updated status for existing orders, right? So will need a server change to also return lastUpdated. And after parameter will need to be based on last updated time rather than created time.